File: Task320/publick/src/main/java/com/publick/service/PostService.java
package com.publick.service;

import com.publick.dto.PostRequestTo;
import com.publick.dto.PostResponseTo;
import com.publick.entity.Issue;
import com.publick.entity.Post;
import com.publick.repository.IssueRepository;
import com.publick.repository.PostRepository;
import com.publick.service.mapper.PostMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class PostService {

    @Autowired
    private PostRepository postRepository;

    @Autowired
    private IssueRepository issueRepository;

    @Autowired
    private PostMapper postMapper;

    public PostResponseTo create(PostRequestTo request) {
        // Validate that issue exists
        Issue issue = issueRepository.findById(request.getIssueId())
                .orElseThrow(() -> new com.publick.exception.NotFoundException("Issue not found with id: " + request.getIssueId()));

        Post post = postMapper.toEntity(request, issue);
        Post saved = postRepository.save(post);
        return postMapper.toResponse(saved);
    }

    public PostResponseTo getById(Long id) {
        Post post = postRepository.findById(id)
                .orElseThrow(() -> new com.publick.exception.NotFoundException("Post not found with id: " + id));
        return postMapper.toResponse(post);
    }

    public List<PostResponseTo> getAll() {
        return postRepository.findAll().stream()
                .map(postMapper::toResponse)
                .collect(Collectors.toList());
    }

    public Page<PostResponseTo> getAll(Pageable pageable) {
        return postRepository.findAll(pageable)
                .map(postMapper::toResponse);
    }

    public PostResponseTo update(Long id, PostRequestTo request) {
        Post existing = postRepository.findById(id)
                .orElseThrow(() -> new com.publick.exception.NotFoundException("Post not found with id: " + id));

        // Validate that issue exists
        Issue issue = issueRepository.findById(request.getIssueId())
                .orElseThrow(() -> new com.publick.exception.NotFoundException("Issue not found with id: " + request.getIssueId()));

        postMapper.updateEntityFromDto(request, existing, issue);
        Post saved = postRepository.save(existing);
        return postMapper.toResponse(saved);
    }

    public void delete(Long id) {
        if (!postRepository.existsById(id)) {
            throw new com.publick.exception.NotFoundException("Post not found with id: " + id);
        }
        postRepository.deleteById(id);
    }
}

File: 351001/Viktor/Task330/discussion/services/post_service.py
from ...publisher.src.core.errors import HttpNotFoundError
from Task330.discussion.domain.repositories.post_repository import CassandraPostRepository

class PostService:
    def __init__(self, repo: CassandraPostRepository):
        self._repo = repo

    async def get_one(self, tweet_id: int, post_id: int):
        post = await self._repo.get_one(tweet_id, post_id)
        if not post:
            raise HttpNotFoundError("Post not found", 40401)
        return post

    async def get_all(self, tweet_id: int, page: int, size: int):
        return await self._repo.get_all(tweet_id, page, size)

    async def create(self, tweet_id: int, content: str):
        return await self._repo.create(tweet_id, content)

    async def update(self, tweet_id: int, post_id: int, content: str):
        return await self._repo.update(tweet_id, post_id, content)

    async def delete(self, tweet_id: int, post_id: int):
        await self._repo.delete(tweet_id, post_id)